client/state/purchases/selectors.js

```diff
@@ -79,7 +79,8 @@ export const getIncludedDomainPurchase = ( state, subscriptionPurchase ) => {
 		sitePurchases,
 		purchase =>
 			( isDomainMapping( purchase ) || isDomainRegistration( purchase ) ) &&
-			includedDomain === purchase.meta
+			includedDomain === purchase.meta &&
+			! getRefundAmount( purchase )
```

Comment:
Thank you for that example; I was clear that this was a bit of an indirect way to determine if the domain registration had been paid, but I couldn't think of a situation where it would generate a false positive. In fact, since AFAIK this function is only used for refund logic, it actually seemed sensible (if a domain reg. has its own refund amount, we shouldn't bundle the refund). I'll take a look at the server response and see if I can add a new property.

Reply:
It also occurs to me that maybe we should have a guard in the display code that prevents showing a plan cost that's greater than the cost of the plan.

Reply:
Not a bad idea, as long as it made sure to compare against the price the user actually paid for the plan (not necessarily the same as the current price).  Otherwise it could cause problems if prices are ever lowered :)